### PR TITLE
LPD-31981 - Address ERC gaps in Headless APIs for productSpecifications

### PR DESCRIPTION
**Pull request by andrea-ale-sbarra**
https://liferay.atlassian.net/browse/LPD-31981

### CODE REVIEW

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Review by alessiorendina — APPROVED** [review]
LGTM, thank you!

**Comment by alessiorendina** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-test-util/src/main/java/com/liferay/commerce/product/test/util/CPTestUtil.java [hunk 1]
```diff
@@ -707,6 +709,20 @@ public static CPOption addCPOption(
 			skuContributor, RandomTestUtil.randomString(), serviceContext);
 	}
 
+	public static CPOptionCategory addCPOptionCategory(long groupId)
+		throws PortalException {
+
+		ServiceContext serviceContext =
+			ServiceContextTestUtil.getServiceContext(groupId);
```
Comment (smottal) [inline modules/apps/commerce/commerce-product-test-util/src/main/java/com/liferay/commerce/product/test/util/CPTestUtil.java:716]: inline this

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/util/v1_0/ProductSpecificationUtil.java [hunk 2]
```diff
@@ -83,21 +100,40 @@ public class ProductSpecificationUtil {
...
-		if (productSpecification.getOptionCategoryId() == null) {
-			if (cpSpecificationOption != null) {
-				return cpSpecificationOption.getCPOptionCategoryId();
-			}
+		if (cpOptionCategory != null) {
+			return cpOptionCategory.getCPOptionCategoryId();
+		}
+
+		cpOptionCategory = cpOptionCategoryService.fetchCPOptionCategory(
+			GetterUtil.get(productSpecification.getOptionCategoryId(), 0));
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/util/v1_0/ProductSpecificationUtil.java:122]: getLong

**Review by smottal — CHANGES_REQUESTED** [review]

File: modules/apps/commerce/commerce-product-test-util/src/main/java/com/liferay/commerce/product/test/util/CPTestUtil.java (hunk 1 above)
Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-product-test-util/src/main/java/com/liferay/commerce/product/test/util/CPTestUtil.java:716]: no it's used twice, I need the variable

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:forward:force

**Comment by andrea-ale-sbarra** [comment]
ci:forward:force

**Comment by andrea-ale-sbarra** [comment]
ci:forward:force

**Comment by andrea-ale-sbarra** [comment]
ci:forward:force

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:forward:force

**Pull request closed without merging**